parsons/bloomerang/bloomerang.py

```diff
@@ -0,0 +1,255 @@
+            self._generate_access_token()
+            headers['Authorization'] = f"Bearer {self.access_token}"
+        else:
+            logger.warning('Initializing API Connector without authorization credentials.')
+        return APIConnector(uri=self.uri, headers=headers)
+
+    def _generate_authorization_code(self):
+        data = {'client_id': self.client_id,
+                'response_type': 'code'}
+        r = requests.post(url=self.uri_auth, json=data)
+        self.authorization_code = r.json().get('code', None)
+
+    def _generate_access_token(self):
+        data = {'client_id': self.client_id,
+                'client_secret': self.client_secret,
+                'grant_type': 'authorization_code',
+                'code': self.authorization_code}
+        r = requests.post(url=self.uri + 'oauth/token', json=data)
+        self.access_token = r.json().get('access_token', None)
+
+    def _base_endpoint(self, endpoint, entity_id=None):
+        url = f'{self.uri}{endpoint}/'
+        if entity_id:
+            url = url + f'{entity_id}/'
+        return url
+
+    def _base_create(self, endpoint, entity_id=None, **kwargs):
+        return self.conn.post_request(url=self._base_endpoint(endpoint, entity_id),
+                                      json=json.dumps({**kwargs}))
+
+    def _base_update(self, endpoint, entity_id=None, **kwargs):
+        return self.conn.put_request(url=self._base_endpoint(endpoint, entity_id),
+                                     json=json.dumps({**kwargs}))
+
+    def _base_get(self, endpoint, entity_id=None):
+        return self.conn.get_request(url=self._base_endpoint(endpoint, entity_id))
+
+    def _base_delete(self, endpoint, entity_id=None):
+        return self.conn.delete_request(url=self._base_endpoint(endpoint, entity_id))
+
+    def create_constituent(self, **kwargs):
+        """
+        `Args:`
+            **kwargs:`
+                Fields to include, e.g., FirstName = 'Rachel'
+        """
+        return self._base_create('constituent', **kwargs)
+
+    def update_constituent(self, constituent_id, **kwargs):
+        """
+        `Args:`
+            constituent_id:
+                Constituent ID to update
+            **kwargs:`
+                Fields to update, e.g., FirstName = 'RJ'
+        """
+        return self._base_update('constituent', entity_id=constituent_id, **kwargs)
+
+    def get_constituent(self, constituent_id):
+        """
+        `Args:`
+            constituent_id:
+                Constituent ID to get fields for
+        `Returns:`
+            A  JSON of the entry or an error.
+        """
+        return self._base_get('constituent', entity_id=constituent_id)
+
+    def delete_constituent(self, constituent_id):
+        """
+        `Args:`
+            constituent_id:
+                Constituent ID to delete
+        """
+        return self._base_delete('constituent', entity_id=constituent_id)
+
+    def get_constituents(self, constituent_ids):
```

Comment:
@eliotst thanks, almost done with revisions. One follow question about your last suggestion: Should the `page_number` parameter be required, or optional? If the latter, what should the behavior be when no `page_number` is provided?

Reply:
Good question. What we have done in other places is if no `page_number` is provided, we have the Connector page through all of pages from the API and return a single `Table` with everything (e.g. [Phone2Action](https://github.com/move-coop/parsons/blob/cc449e31a5e76bc3bd451d43cd6e3902769cba4b/parsons/phone2action/p2a.py#L35-L54)).. but that's a non-trivial change, and definitely more nice-to-have. I think for this PR it's okay to just default to the first page (`0` or `1` depending on what the API considers "first").

Reply:
Okay, I'll keep it simple and default to the first page for this PR

Reply:
> it might be worth testing whether or not we can pass a `take` higher than the default 50 items. 50 is a little small for a page, especially if someone is trying to download a lot of constituents.

It actually seems like 50 is the maximum for this API

Reply:
Quick follow up - given this discussion about pagination, do you think we should add some guidance to the How To Build a Connector docs? It seems like this would be a common issue.